index.js

```diff
@@ -107,7 +108,7 @@ module.exports = function(md, options) {
           headings.push(buffer);
         }
       }
-      buffer = "<li><a href=\"#" + options.slugify(heading.content) + "\">" + heading.content + "</a>";
+      buffer = "<li><a href=\"#" + options.slugify(heading.content) + "\">" + options.filter(heading.content) + "</a>";
```

Comment:
How about not having a default at all? You could just:
`typeof options.filter === "function" ? options.filter.call(heading.content) : heading.content`